src/sentry/models/integrations/organization_integrity_backfill_mixin.py

```diff
@@ -0,0 +1,26 @@
+# Mixin intended to be attached to Model classes that belong to an opposing silo.
+# Allows looking up and storing the organization_id and integration_id of an external
+# OrganizationIntegration, and powers backfill associated with each.
+from __future__ import annotations
+
+from typing import Any
+
+from sentry.services.hybrid_cloud.integration import integration_service
+
+
+class OrganizationIntegrityBackfillMixin:
+    organization_integration_id: Any
+    organization_id: Any
+    integration_id: Any
+
+    def save(self, *args, **kwds) -> None:
+        if self.organization_id is None or self.integration_id is None:
+            # Find the original org integration instance, backfill in the identifiers.
+            org_integrations = integration_service.get_organization_integrations(
+                org_integration_ids=[self.organization_integration_id],
+            )
```

Comment:
Why do we do this instead of just using `.organization_integration`?

Reply:
Because of tests that assert the silo boundary right now.